components/script/dom/worker.rs

```diff
@@ -100,6 +103,10 @@ impl Worker {
                           data: StructuredCloneData) {
         let worker = address.to_temporary().root();
 
+        if worker.r().closing.get() {
```

Comment:
[The spec](https://html.spec.whatwg.org/multipage/#terminate-a-worker) specifies to empty the event/task queue associated with the worker as to not deliver further events to the `onmessage` handler. Without a synchronous way to affect the `DedicatedWorkerGlobalScope`, I resorted to these short-circuiting checks to prevent messages from being delivered. I keep thinking about this because it's kind of ugly. Does someone have any perspective on a cleaner way to achieve that behavior?